meta/MetaKeyHasher.cpp

```diff
@@ -191,6 +191,22 @@ static bool operator==(
         a.label == b.label;
 }
 
+static bool operator==(
+        _In_ const sai_my_sid_entry_t& a,
+        _In_ const sai_my_sid_entry_t& b)
+{
+    // SWSS_LOG_ENTER(); // disabled for performance reasons
+
+    bool part = a.switch_id == b.switch_id &&
+        a.vr_id == b.vr_id &&
+        a.locator_block_len == b.locator_block_len &&
+        a.locator_node_len == b.locator_node_len &&
+        a.function_len == b.function_len &&
+        a.args_len == b.args_len;
+
+    return part && memcmp(a.sid, b.sid, sizeof(a.sid)) == 0;
```

Comment:
what type is a.sid ? maybe better would be to introduce new compare == opeartaor for that type ?

Reply:
sid is an array of standard type (uint8).

sid - sai_ip6_t